Updated doc pages for Docker, Kafka, Nginx ST's

yasar-sumologic opened the pull request:
## Purpose of this pull request

This pull request adds Docker, Kafka, Nginx STs


## Select the type of change
<!-- What types of changes does your code introduce? Select the checkbox after clicking "Create pull request" button. -->

- [ ] Minor Changes - Typos, formatting, slight revisions
- [ ] Update Content - Revisions, updating sections
- [X] New Content - New features, sections, pages, tutorials
- [ ] Site and Tools - .clabot, version updates, maintenance, dependencies, new packages for the site (Docusaurus, Gatsby, React, etc.)

## Ticket (if applicable)

<!-- enter your Jira, Asana, or GitHub ticket number -->

sumoanema commented on docs/send-data/opentelemetry-collector/remote-management/source-templates/docker.md:
```diff
@@ -0,0 +1,84 @@
+- **Description**. Description for the source template.	
+- **Docker Event log location**. Enter the path of the JSON file generated through the command in the prerequisite section.
+- **Endpoint**. Address to reach the desired Docker daemon (default: `unix:///var/run/docker.sock`).
+- **Excluded Image List**. A list of strings, [regexes](https://golang.org/pkg/regexp/), or [globs](https://github.com/gobwas/glob) whose referent container image names will not be among the queried containers for scrapping metrics. Learn more about [*excluded_images*](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/dockerstatsreceiver/README.md#configuration) ...
```
Add Fields/Metadata user input

yasar-sumologic replied:
fixed it.

kimsauce commented on docs/send-data/opentelemetry-collector/remote-management/source-templates/docker.md:
```diff
@@ -0,0 +1,85 @@
+
+### Step 2: Configure the source template
+
+In this step, you will configure the yaml required for Docker Collection. Below are the inputs required for configuration :
```
```suggestion
In this step, you will configure the yaml required for Docker Collection. Below are the inputs required for configuration:
```

yasar-sumologic replied:
fixed it.

kimsauce commented on docs/send-data/opentelemetry-collector/remote-management/source-templates/nginx.md:
```diff
@@ -0,0 +1,95 @@
+		
+### Step 2: Configure the source template
+		
+In this step, you will configure the yaml required for Nginx Collection. Below are the inputs required for configuration :
```
```suggestion
In this step, you will configure the yaml required for Nginx collection. Below are the inputs required for configuration:
```

yasar-sumologic replied:
fixed it.

kimsauce commented on docs/send-data/opentelemetry-collector/remote-management/source-templates/nginx.md:
```diff
@@ -0,0 +1,95 @@
+- **Description**. Description for the source template.	
+- **Endpoint**. The URL of the httpd status endpoint (default: `http://localhost:80/status`).
+- **Path to Nginx access Log file**. Enter the path to the Access log file for your Nginx instance.
+- **Path to Nginx Error Log file**. Enter the path to the error log file for your Nginx instance.
```
```suggestion
- **Path to Nginx error Log file**. Enter the path to the error log file for your Nginx instance.
```

yasar-sumologic replied:
fixed it.

kimsauce commented on docs/send-data/opentelemetry-collector/remote-management/source-templates/docker.md:
```diff
@@ -0,0 +1,85 @@
+
+This section provides instructions for configuring metrics and log collection for the Sumo Logic app for Docker.
+
+#### For Metric collection
```
```suggestion
#### For metrics collection
```

yasar-sumologic replied:
fixed it.

kimsauce commented on docs/send-data/opentelemetry-collector/remote-management/source-templates/docker.md:
```diff
@@ -0,0 +1,85 @@
+
+Metrics are collected through the [Docker Stats Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/dockerstatsreceiver/README.md) of OpenTelemetry. This requires Docker API version 1.22+ and only Linux is supported.
+
+#### For Log collection
```
```suggestion
#### For logs collection
```

yasar-sumologic replied:
fixed it.

kimsauce commented on docs/send-data/opentelemetry-collector/remote-management/source-templates/docker.md:
````diff
@@ -0,0 +1,85 @@
+```
+docker events -f 'type=container' --format '{{json .}}' > <PATH_TO_JSON> & disown
+```
+Path to this JSON file will be required in the [next step](#step-2-configure-integration), where events are sent to Sumo Logic through a filelog receiver and seen as part of the **Docker - Overview** dashboard. Also, you can add additional parameters to this command to send events for specific containers. [Learn more](https://docs.docker.com/engine/reference/commandline/events/).
````
```suggestion

The path to this JSON file will be required in the [next step](#step-2-configure-integration), where events are sent to Sumo Logic through a filelog receiver and seen as part of the **Docker - Overview** dashboard. Also, you can add additional parameters to this command to send events for specific containers. [Learn more](https://docs.docker.com/engine/reference/commandline/events/).
```

yasar-sumologic replied:
fixed it.

sumoanema commented on docs/send-data/opentelemetry-collector/remote-management/source-templates/docker.md:
```diff
@@ -0,0 +1,85 @@
+- **Docker Event log location**. Enter the path of the JSON file generated through the command in the prerequisite section.
+- **Endpoint**. Address to reach the desired Docker daemon (default: `unix:///var/run/docker.sock`).
+- **Excluded Image List**. A list of strings, [regexes](https://golang.org/pkg/regexp/), or [globs](https://github.com/gobwas/glob) whose referent container image names will not be among the queried containers for scrapping metrics. Learn more about [*excluded_images*](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/dockerstatsreceiver/README.md#configuration) ...
+- **Fields/Metadata**. You can provide any customer fields to be tagged with the data collected. By default sumo tags `_sourceCategory` with the value otel/kafka.
```
otel/kafka ? should be otel/docker

yasar-sumologic replied:
fixed it.

kimsauce commented on docs/send-data/opentelemetry-collector/remote-management/source-templates/docker.md:
```diff
@@ -0,0 +1,85 @@
+
+<img src={useBaseUrl('img/integrations/containers-orchestration/docker-icon.png')} alt="Thumbnail icon" width="100"/> <img src={useBaseUrl('img/send-data/otel-color.svg')} alt="Thumbnail icon" width="30"/>
+
+Docker source template creates an OpenTelemetry configuration that can be pushed to a remotely managed OpenTelemetry collector (abbreviated as otelcol). By creating this source template and pushing the config to the appropriate OpenTelemetry agent you can ensure collection of logs and metrics of Docker to Sumo Logic.
```
```suggestion
The Docker source template creates an OpenTelemetry configuration that can be pushed to a remotely managed OpenTelemetry collector (abbreviated as otelcol). By creating this source template and pushing the config to the appropriate OpenTelemetry agent, you can ensure collection of logs and metrics of Docker to Sumo Logic.
```

kimsauce commented on docs/send-data/opentelemetry-collector/remote-management/source-templates/docker.md:
```diff
@@ -0,0 +1,85 @@
+
+## Prerequisites
+
+This section provides instructions for configuring metrics and log collection for the Sumo Logic app for Docker.
```
```suggestion
This section provides instructions for configuring metrics and log collection for the Sumo Logic Docker app.
```

kimsauce commented on docs/send-data/opentelemetry-collector/remote-management/source-templates/docker.md:
```diff
@@ -0,0 +1,85 @@
+
+#### For logs collection
+
+To collect the Docker container event logs, the following command needs to be executed on the host machine and needs to be kept running, for monitoring all the Docker container-related events. The following command also needs a JSON file path where these container events can be dumped.
```
```suggestion
To collect Docker container event logs, execute the following command on the host machine and keep it running to monitor all Docker container-related events. The command requires a JSON file path where these container events will be stored.
```

kimsauce commented on docs/send-data/opentelemetry-collector/remote-management/source-templates/docker.md:
```diff
@@ -0,0 +1,85 @@
+- **Docker Event log location**. Enter the path of the JSON file generated through the command in the prerequisite section.
+- **Endpoint**. Address to reach the desired Docker daemon (default: `unix:///var/run/docker.sock`).
+- **Excluded Image List**. A list of strings, [regexes](https://golang.org/pkg/regexp/), or [globs](https://github.com/gobwas/glob) whose referent container image names will not be among the queried containers for scrapping metrics. Learn more about [*excluded_images*](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/dockerstatsreceiver/README.md#configuration) ...
+- **Fields/Metadata**. You can provide any customer fields to be tagged with the data collected. By default sumo tags `_sourceCategory` with the value otel/docker.
```
```suggestion
- **Fields/Metadata**. You can provide any customer fields to be tagged with the data collected. By default, Sumo Logic tags `_sourceCategory` with the value otel/docker.
```

kimsauce commented on docs/send-data/opentelemetry-collector/remote-management/source-templates/docker.md:
````diff
@@ -0,0 +1,85 @@
+```
+docker events -f 'type=container' --format '{{json .}}' > <PATH_TO_JSON> & disown
+```
+The path to this JSON file will be required in the [next step](#step-2-configure-integration), where events are sent to Sumo Logic through a filelog receiver and seen as part of the **Docker - Overview** dashboard. Also, you can add additional parameters to this command to send events for specific containers. [Learn more](https://docs.docker.com/engine/reference/commandline/events/).
````
This link doesn't work: `[next step](#step-2-configure-integration)`

yasar-sumologic replied:
removed that link.

kimsauce commented on docs/send-data/opentelemetry-collector/remote-management/source-templates/kafka.md:
```diff
@@ -0,0 +1,87 @@
+
+<img src={useBaseUrl('img/integrations/containers-orchestration/kafka.png')} alt="icon" width="90"/> <img src={useBaseUrl('img/send-data/otel-color.svg')} alt="Thumbnail icon" width="45"/>
+
+Kafka source template creates an OpenTelemetry configuration that can be pushed to a remotely managed OpenTelemetry collector (abbreviated as otelcol). By creating this source template and pushing the config to the appropriate OpenTelemetry agent you can ensure collection of logs and metrics of Kafka to Sumo Logic.
```
```suggestion
The Kafka source template generates an OpenTelemetry configuration that can be sent to a remotely managed OpenTelemetry collector (otelcol). By creating this source template and pushing the configuration to the appropriate OpenTelemetry agent, you can ensure the collection of Kafka logs and metrics in Sumo Logic.
```

kimsauce commented on docs/send-data/opentelemetry-collector/remote-management/source-templates/kafka.md:
```diff
@@ -0,0 +1,87 @@
+
+- **`sumo.datasource`**. Fixed value of **kafka**.
+- **`messaging.system`**. Fixed value of **kafka**.
+- **`deployment.environment`**. This is a user-configured field set at the time of collector installation. It identifies the environment where the kafka env resides, such as `dev`, `prod`, or `qa`.
```
```suggestion
- **`deployment.environment`**. This is a user-configured field set at the time of collector installation. It identifies the environment where the Kafka env resides, such as `dev`, `prod`, or `qa`.
```

kimsauce commented on docs/send-data/opentelemetry-collector/remote-management/source-templates/kafka.md:
```diff
@@ -0,0 +1,87 @@
+
+### For metrics collection
+
+Kafka metrics [receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/kafkametricsreceiver) collects Kafka metrics (brokers, topics, partitions, and consumer groups) from the Kafka server. This app has been tested with following Kafka versions: 2.X and 3.X
```
```suggestion
The Kafka metrics [receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/kafkametricsreceiver) collects Kafka metrics (brokers, topics, partitions, and consumer groups) from the Kafka server. This app has been tested with following Kafka versions: 2.x and 3.x.
```

kimsauce commented on docs/send-data/opentelemetry-collector/remote-management/source-templates/kafka.md:
```diff
@@ -0,0 +1,87 @@
+
+### For logs collection
+
+Configure logging in Kafka: By default, Kafka logs (`server.log` and `controller.log`) are stored in the directory called `/opt/Kafka/kafka_<VERSION>/logs`. Make a note of this logs directory.
```
```suggestion
In this section, you'll configure logging in Kafka. By default, Kafka logs (`server.log` and `controller.log`) are stored in the directory called `/opt/Kafka/kafka_<VERSION>/logs`. Make a note of this logs directory.
```

kimsauce commented on docs/send-data/opentelemetry-collector/remote-management/source-templates/kafka.md:
```diff
@@ -0,0 +1,87 @@
+
+### Step 2: Configure the source template
+
+In this step, you will configure the yaml required for Kafka Collection. Below are the inputs required for configuration :
```
```suggestion
In this step, you will configure the yaml required for Kafka collection. Below are the inputs required for configuration:
```

kimsauce commented on docs/send-data/opentelemetry-collector/remote-management/source-templates/kafka.md:
```diff
@@ -0,0 +1,87 @@
+
+- **Name**. Name of the source template.
+- **Description**. Description for the source template.	
+- **Server File log Path**. Enter the path to the Server log file for your Kafka instance.
+- **Controller file log path**. Enter the path to the Controller log file for your Kafka instance.
```
```suggestion
- **Server File log Path**. Enter the path to the server log file for your Kafka instance.
- **Controller file log path**. Enter the path to the controller log file for your Kafka instance.
```

kimsauce commented on docs/send-data/opentelemetry-collector/remote-management/source-templates/kafka.md:
```diff
@@ -0,0 +1,87 @@
+- **Server File log Path**. Enter the path to the Server log file for your Kafka instance.
+- **Controller file log path**. Enter the path to the Controller log file for your Kafka instance.
+- **Endpoint**. The URL of the broker endpoint (default: `localhost:9092`).
+- **Fields/Metadata**. You can provide any customer fields to be tagged with the data collected. By default sumo tags `_sourceCategory` with the value otel/kafka user needs to provide the value for `webengine.cluster.name`.
```
```suggestion
- **Fields/Metadata**. You can provide any customer fields to be tagged with the data collected. By default, Sumo Logic tags `_sourceCategory` with the value otel/Kafka user needs to provide the value for `webengine.cluster.name`.
```

kimsauce commented on docs/send-data/opentelemetry-collector/remote-management/source-templates/nginx.md:
```diff
@@ -0,0 +1,95 @@
+
+<img src={useBaseUrl('img/integrations/web-servers/nginx.png')} alt="Thumbnail icon" width="100"/> <img src={useBaseUrl('img/send-data/otel-color.svg')} alt="Thumbnail icon" width="30"/>
+
+Nginx source template creates an OpenTelemetry configuration that can be pushed to a remotely managed OpenTelemetry collector (abbreviated as otelcol). By creating this source template and pushing the config to the appropriate OpenTelemetry agent you can ensure collection of logs and metrics of Nginx to Sumo Logic.
```
```suggestion
The Nginx source template generates an OpenTelemetry configuration that can be sent to a remotely managed OpenTelemetry collector (otelcol). By creating this source template and pushing the configuration to the appropriate OpenTelemetry agent, you can ensure the collection of Nginx logs and metrics in Sumo Logic.
```

kimsauce commented on docs/send-data/opentelemetry-collector/remote-management/source-templates/kafka.md:
```diff
@@ -0,0 +1,87 @@
+
+You can follow the below steps to set a remotely managed OpenTelemetry collector and push the source template to it.
+
+### Step 1: Step 1: Set up remotely managed OpenTelemetry collector
```
```suggestion
### Step 1: Set up remotely managed OpenTelemetry collector
```

kimsauce commented on docs/send-data/opentelemetry-collector/remote-management/source-templates/nginx.md:
```diff
@@ -0,0 +1,95 @@
+### For logs collection
+		
+Configure the Nginx access and error log files:
+1. Configure the logging of access logs and error logs via the instructions described in their [documentation](https://docs.nginx.com/nginx/admin-guide/monitoring/logging/).
```
```suggestion
1. Configure the logging of access logs and error logs by following the instructions in their [documentation](https://docs.nginx.com/nginx/admin-guide/monitoring/logging/).
```

kimsauce reviewed (changes requested):
Had some more changes!

yasar-sumologic commented:
> Had some more changes!

Hi @kimsauce 
I've addressed your feedback on PR and believe it's ready for re-review; could you please take another look?
